contracts/Strategy.sol

```diff
@@ -53,80 +222,624 @@ contract Strategy is BaseStrategy {
             uint256 _debtPayment
         )
     {
-        // TODO: Do stuff here to free up any returns back into `want`
-        // NOTE: Return `_profit` which is value generated by all positions, priced in `want`
-        // NOTE: Should try to free up at least `_debtOutstanding` of underlying position
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            accrueInterest(i);
+            depositKashiInMasterChef(i); // claim and deposit loose
+        }
+
+        sell();
+
+        uint256 assets = estimatedTotalAssets();
+        uint256 wantBal = balanceOfWant();
+
+        uint256 debt = vault.strategies(address(this)).totalDebt;
+
+        if (assets >= debt) {
+            _profit = assets.sub(debt);
+        } else {
+            _loss = debt.sub(assets);
+        }
+
+        _debtPayment = _debtOutstanding;
+        uint256 amountToFree = _debtPayment.add(_profit);
+
+        if (amountToFree > 0 && wantBal < amountToFree) {
+            (uint256 newLoose, ) = liquidatePosition(amountToFree.sub(wantBal));
+
+            // if we didnt free enough money, prioritize paying down debt before taking profit
+            if (newLoose < amountToFree) {
+                if (newLoose <= _debtPayment) {
+                    _profit = 0;
+                    _loss += _debtPayment.sub(newLoose);
+                    _debtPayment = newLoose;
+                } else {
+                    _profit = newLoose.sub(_debtPayment);
+                }
+            }
+        }
     }
 
     function adjustPosition(uint256 _debtOutstanding) internal override {
-        // TODO: Do something to invest excess `want` tokens (from the Vault) into your positions
-        // NOTE: Try to adjust positions so that `_debtOutstanding` can be freed up on *next* harvest (not immediately)
+        if (emergencyExit) {
+            return;
+        }
+
+        uint256 wantBalance = balanceOfWant();
+
+        uint256 shares = 0;
+
+        if (wantBalance > dustThreshold) {
+            (, shares) = depositInBento(wantBalance);
+        }
+
+        uint256 sharesInBento = sharesInBento();
+
+        if (sharesInBento > wantToBentoShares(dustThreshold, false)) {
```

Comment:
I see this is the only place where `wantToBentoShares` is called with roundup=false

given that the default value is true always i think this may better be refactored to `wantToBentoSharesRoundup(dustThreshold)` and use `wantToBentoShares(amount)` with true since it cleans up all the rest of the code